providers/src/airflow/providers/elasticsearch/log/es_task_handler.py

```diff
@@ -266,15 +271,18 @@ def _render_log_id(self, ti: TaskInstance | TaskInstanceKey, try_number: int) ->
                 data_interval_end = data_interval[1].isoformat()
             else:
                 data_interval_end = ""
-            execution_date = dag_run.execution_date.isoformat()
+            logical_date = (
+                dag_run.logical_date.isoformat() if AIRFLOW_V_3_0_PLUS else dag_run.execution_date.isoformat()
+            )
```

Comment:
```suggestion
            logical_date = dag_run.logical_date.isoformat()
```

`dag_run.logical_date` is available on Airflow 2.

Reply:
Changed it